controllers/argocd/status.go

```diff
@@ -212,3 +217,43 @@ func (r *ReconcileArgoCD) reconcileStatusServer(cr *argoprojv1a1.ArgoCD) error {
 	}
 	return nil
 }
+
+// reconcileStatusURL will ensure that the URL status is updated for the given ArgoCD.
+func (r *ReconcileArgoCD) reconcileStatusURL(cr *argoprojv1a1.ArgoCD) error {
+
+	if cr.Spec.Server.Route.Enabled {
+		if IsRouteAPIAvailable() {
+			route := newRouteWithSuffix("server", cr)
+			if argoutil.IsObjectFound(r.Client, cr.Namespace, route.Name, route) {
+				if route.Status.Ingress == nil {
+					status := "Pending"
+					cr.Status.URL = status
+					cr.Status.Phase = status
+					return r.Client.Status().Update(context.TODO(), cr)
+				} else {
+					cr.Status.URL = route.Spec.Host
+					cr.Status.Phase = "Available"
+					return r.Client.Status().Update(context.TODO(), cr)
+				}
+			}
+			return nil
+		}
+		log.Info("Routes not available in non-OpenShift environments, please use Ingresses instead")
```

Comment:
Could you put this in an else block? Otherwise this statement would get logged even when RouteAPI is available

Reply:
instead use 
if ! IsRouteAPIAvailable() {
       log.Info("Routes not available in non-OpenShift environments, please use Ingresses instead")
}

route := newRouteWithSuffix("server", cr)
.....
.......

Reply:
else block is not needed :)

Reply:
That sounds good too
as long as we return after logging